lib/Doctrine/DBAL/Driver/Mysqli/MysqliConnection.php

```diff
@@ -263,4 +264,35 @@ public function ping()
     {
         return $this->_conn->ping();
     }
+
+    /**
+     * Stablish a secure connection
+     *
+     * @param array $params
+     * @throws MysqliException
+     */
+    private function setSecureConnection(array $params)
+    {
+        if (! isset($params['ssl_key']) &&
+            ! isset($params['ssl_cert']) &&
+            ! isset($params['ssl_ca']) &&
+            ! isset($params['ssl_capath']) &&
+            ! isset($params['ssl_cipher'])
+        ) {
+            return;
+        }
+
+        if (! isset($params['ssl_key']) || ! isset($params['ssl_cert'])) {
+            $msg = "\"ssl_key\" and \"ssl_cert\" parameters are mandatory when using secure connection parameters.";
+            throw new MysqliException($msg);
```

Comment:
Please don't do unnecessary string escaping. You can easily avoid that by using single quotes:

```php
throw new MysqliException(
    '"ssl_key" and "ssl_cert" parameters are mandatory when using secure connection parameters.'
);
```